features/config/service/impl/src/main/java/org/opennms/features/config/service/impl/ConfigurationManagerServiceImpl.java

```diff
@@ -165,14 +171,27 @@ public void updateConfiguration(String configName, String configId, JsonAsString
 
     @Override
     public Optional<JSONObject> getJSONConfiguration(final String configName, final String configId) throws IOException {
-        return configStoreDao.getConfig(configName, configId);
+        Optional<JSONObject> configObj = configStoreDao.getConfig(configName, configId);
+        if (configObj.isEmpty()) {
+            return configObj;
+        }
+        // try to fill default value or empty signature
+        Optional<ConfigDefinition> def = configStoreDao.getConfigDefinition(configName);
+        if (def.isPresent()) {
+            String schemaName = def.get().getMetaValue(ConfigDefinition.TOP_LEVEL_ELEMENT_NAME_TAG);
+            if (schemaName == null) { // assume if top element name is null, the top schema name is configName
+                schemaName = configName;
+            }
+            OpenAPIConfigHelper.fillWithDefaultValue(def.get().getSchema(), schemaName, configObj.get());
+        }
+        return configObj;
     }
 
     @Override
     public String getJSONStrConfiguration(String configName, String configId) throws IOException, IllegalArgumentException {
```

Comment:
I wonder if we shouldn't use here Optional<String> getJSONStrConfiguration instead of throwing a FileNotFoundException? This would be consistent to getJSONConfiguration()